projects/VitalRecord/VitalRecord_util.cs

```diff
@@ -109,6 +109,18 @@ protected Observation GetObservation(string code)
             return null;
         }
 
+        /// <summary>
+        /// Helper to support retrieval of matching Observations in cases where there can be multiple matches.
+        /// </summary>
+        /// <param name="code">The code to identify the type of Observation.</param>
+        /// <returns>A list of matching Observations, or an empty list if no matches are found.</returns>
+        protected List<Observation> GetObservations(string code)
+        {
+            return Bundle.Entry.Where(e => e.Resource is Observation obs && CodeableConceptToDict(obs.Code)["code"] == code)
```

Comment:
`CodeableConceptToDict` only looks at the first code in the `obs.code.coding` list of codes. Not sure if that is a problem in this context but you could write this to support the general case (more than one code where the match isn't the first) fairly easily if needed.

Reply:
Yeah, I feel like this assumption (there will only ever be one code) has bitten us at times, but for this case, are there observations that have more than one code specifying the type of observation? Maybe worth a separate spike to look for this pattern and generalize across the entire code base.